src/filters/filters.jl

```diff
@@ -287,12 +291,15 @@ function Base.delete!(filters::FilterPipeline, ::Type{F}) where {F<:Filter}
     API.h5p_remove_filter(filters.plist, filterid(F))
     return filters
 end
-function Base.append!(filters::FilterPipeline, extra)
+function Base.append!(filters::FilterPipeline, extra::Union{AbstractVector{<:Filter}, NTuple{N, Filter} where N})
     for filter in extra
         push!(filters, filter)
     end
     return filters
 end
+function Base.append!(filters::FilterPipeline, extra::NTuple{N, Integer}) where N
```

Comment:
I guess this is only added to support existing use cases.

Would it make sense to wrap this is a deprecation warning and force users to manually wrap things in `UnkownFilter`.

I'm leaning towards not doing that and just leaving this as in the PR?  Just a thought.

Reply:
We have already deprecated the upstream interface, so I think that warning should be sufficient. Keeping this just prevents the rather nasty stack overflow situation.

Reply:
This should've been in reference to the addition
```
function Base.push!(p::FilterPipeline, f::Tuple)
```

Reply:
I deprecated all the `Tuple` based methods. The tests give the following warnings.

```
┌ Warning: `filter=(307, 0)` keyword option is deprecated, use `filters=(307, 0)` instead
│   caller = setproperty!(p::HDF5.DatasetCreateProperties, name::Symbol, val::Tuple{Int32, Int64}) at properties.jl:47
└ @ HDF5 C:\Users\kittisopikulm\.julia\dev\HDF5\src\properties.jl:47
┌ Warning: `append!(filters::FilterPipeline, extra::NTuple{N, Integer}) where N` is deprecated, use `append!(filters, [UnknownFilter(extra...)])` instead.
│   caller = set_filters! at properties.jl:403 [inlined]
└ @ Core C:\Users\kittisopikulm\.julia\dev\HDF5\src\properties.jl:403
```